### PR TITLE
catch CertificateError in tls verification

### PR DESCRIPTION
**Pull request by lucywang000**
This should fix https://github.com/scrapy/scrapy/issues/3029 (downlader errors when certificate is issued for ips instead of domains)

### CODE REVIEW

**Comment by lucywang000** [comment]
The tests only fails when "TOXENV=jessie", I'll see how to fix that.

**Comment by curita** [comment]
Oops, sorry about that^, misclicked the reviewers tag.

**Comment by dangra** [comment]
LGTM. We can release it as 1.4.x bugfix.

File: scrapy/core/downloader/tls.py [hunk 1]
```diff
@@ -65,7 +73,7 @@ def _identityVerifyingInfoCallback(self, connection, where, ret):
             elif where & SSL_CB_HANDSHAKE_DONE:
                 try:
                     verifyHostname(connection, self._hostnameASCII)
-                except VerificationError as e:
+                except verification_errors as e:
```
Comment (kmike) [inline scrapy/core/downloader/tls.py]: Could you please update docstring of ScrapyClientTLSOptions? It mentions VerificationError and ValueError, but not CertificateError.

Comment (lucywang000) [reply scrapy/core/downloader/tls.py]: +1 & done

**Comment by dangra** [comment]
thanks!

**Comment by lucywang000** [comment]
Thanks everyone for the reviews !

**Pull request merged**